pkg/mqtrigger/mqtrigger.go

```diff
@@ -40,10 +44,68 @@ func Start(logger *zap.Logger, routerUrl string) error {
 	// Message queue type: nats is the only supported one for now
 	mqType := os.Getenv("MESSAGE_QUEUE_TYPE")
 	mqUrl := os.Getenv("MESSAGE_QUEUE_URL")
+
+	// For authentication with message queue
+	secrets, err := readSecrets(logger)
+
+	if err != nil {
+		panic(err)
+	}
+
 	mqCfg := messageQueue.MessageQueueConfig{
-		MQType: mqType,
-		Url:    mqUrl,
+		MQType:  mqType,
+		Url:     mqUrl,
+		Secrets: secrets,
 	}
 	messageQueue.MakeMessageQueueTriggerManager(logger, fissionClient, routerUrl, mqCfg)
 	return nil
 }
+
+func getCurrentNamespace() string {
+	data, err := ioutil.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/namespace")
+
+	namespace := strings.TrimSpace(string(data))
+	if err == nil && len(namespace) > 0 {
+		return namespace
+	}
+
+	return "default"
+}
+
+func readSecrets(logger *zap.Logger) (map[string][]byte, error) {
+	secretsPath := "/etc/secrets"
+	secrets := make(map[string][]byte)
+
+	// return if no secrets exist
+	if _, err := os.Stat(secretsPath); os.IsNotExist(err) {
+		return secrets, nil
+	}
+
+	secretFiles, err := ioutil.ReadDir(secretsPath)
+
+	if err != nil {
+		return secrets, err
+	}
+
+	for _, secretFile := range secretFiles {
+
+		fileName := secretFile.Name()
+
+		// /etc/secrets contain some hidden directories (like .data)
+		// ignore them
+		if !secretFile.IsDir() && !strings.HasPrefix(fileName, ".") {
+			logger.Info(fmt.Sprintf("Reading secret from %#v", fileName))
+			filePath := path.Join(secretsPath, fileName)
+			secret, fileReadErr := ioutil.ReadFile(filePath)
+
+			if fileReadErr != nil {
+				return secrets, fileReadErr
+			}
+
+			secrets[fileName] = secret
+		}
+
```

Comment:
nitpick: some of blank lines are not really necessary, please remove them. For example,
```
secretFiles, err := ioutil.ReadDir(secretsPath)

if err != nil {
	return secrets, err
}
```

Reply:
Done